src/linux.rs

```diff
@@ -0,0 +1,142 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::{io::Error, net::IpAddr};
+
+use neli::{
+    attr::Attribute,
+    consts::{
+        nl::{NlmF, NlmFFlags},
+        rtnl::{
+            Arphrd, IffFlags, Ifla, RtAddrFamily, RtScope, RtTable, Rta, Rtm, RtmFFlags, Rtn,
+            Rtprot,
+        },
+        socket::NlFamily,
+    },
+    nl::{NlPayload, Nlmsghdr},
+    rtnl::{Ifinfomsg, Rtattr, Rtmsg},
+    socket::NlSocketHandle,
+    types::RtBuffer,
+};
+
+use crate::default_err;
+
+const fn family(remote: &IpAddr) -> RtAddrFamily {
+    if remote.is_ipv4() {
+        RtAddrFamily::Inet
+    } else {
+        RtAddrFamily::Inet6
+    }
+}
+
+const fn addr_len(remote: &IpAddr) -> u8 {
+    if remote.is_ipv4() {
+        32
+    } else {
+        128
+    }
+}
+
+fn addr_bytes(remote: &IpAddr) -> Vec<u8> {
+    match remote {
+        IpAddr::V4(ip) => ip.octets().to_vec(),
+        IpAddr::V6(ip) => ip.octets().to_vec(),
+    }
+}
+
+pub fn interface_and_mtu_impl(remote: IpAddr) -> Result<(String, usize), Error> {
+    // Create a netlink socket
+    let mut socket = NlSocketHandle::connect(NlFamily::Route, None, &[])?;
+
+    // Send RTM_GETROUTE message to retrieve the route for the destination IP
+    let mut route_msg = Rtmsg {
+        rtm_family: family(&remote),
+        rtm_dst_len: addr_len(&remote),
+        rtm_src_len: 0,
+        rtm_tos: 0,
+        rtm_table: RtTable::Default,
+        rtm_protocol: Rtprot::Unspec,
+        rtm_scope: RtScope::Universe,
+        rtm_type: Rtn::Unicast,
+        rtm_flags: RtmFFlags::empty(),
+        rtattrs: RtBuffer::new(),
+    };
+    let rta = Rtattr::new(None, Rta::Dst, addr_bytes(&remote)).map_err(|_| default_err())?;
+    route_msg.rtattrs.push(rta);
+    let route_request = Nlmsghdr::new(
+        None,
+        Rtm::Getroute,
+        NlmFFlags::new(&[NlmF::Request, NlmF::Match]),
+        None,
+        None,
+        NlPayload::Payload(route_msg),
+    );
+    socket.send(route_request).or(Err(default_err()))?;
+
+    // Receive the response to RTM_GETROUTE
+    let response: Nlmsghdr<Rtm, Rtmsg> = socket
+        .recv()
+        .map_err(|_| default_err())?
+        .ok_or_else(default_err)?;
+    let route_attrs = &response.get_payload().map_err(|_| default_err())?.rtattrs;
+    let mut ifindex = None;
+    for attr in route_attrs.iter() {
+        if let Ok(index) = attr.get_payload_as::<i32>() {
+            if attr.rta_type == Rta::Oif {
+                ifindex = Some(index);
+                break;
+            }
+        }
+    }
+    let ifindex = ifindex.ok_or_else(default_err)?;
+    println!("Interface Index: {ifindex}");
```

Comment:
!!!

Reply:
Yeah, the Linux code is still not working, hence this.